src/modules/ngFormLib/controls/common/FormControlService.js

```diff
@@ -185,19 +185,31 @@ mod.provider('formControlService', function() {
         labelElem.prepend(service.translate(labelText));
       },
 
-      addInputGroup: function(inputElem, inputGroupPrefix, inputGroupSuffix) {
-        if (inputGroupPrefix || inputGroupSuffix) {
-          inputElem.wrap('<div class="input-group">'); // This should be the 'control-row' element//wrap('<div class="input-group">');
-          var wrapper = inputElem.parent();
+      addInputGroup: function(inputElem, attr) {
+
+        const inputGroupMapping = {
+          inputPrefix: {className: 'input-group-addon', prepend: true},
+          inputSuffix: {className: 'input-group-addon', prepend: false},
+          inputButtonPrefix: {className: 'input-group-btn', prepend: true},
+          inputButtonSuffix: {className: 'input-group-btn', prepend: false}
+        };
+
+        if (attr.inputPrefix || attr.inputSuffix || attr.inputButtonPrefix || attr.inputButtonSuffix) {
+          inputElem.wrap('<div class="input-group">');//inputElem.parent(); // This should be the 'control-row' element//wrap('<div class="input-group">');
+          let wrapper = inputElem.parent();
+
+          Object.keys(inputGroupMapping).forEach((inputGroupName) => {
```

Comment:
I'm a bit concerned about Object.keys(), as IIRC PhantomJS doesn't understand it and you need to use a Babel transform to make it work for PhantomJS. But the tests have passed, so I'm surprised.